docs/examples/promise/02_listen_to_blocks/index.js

```diff
@@ -9,14 +9,20 @@ async function main () {
 
   // Subscribe to the new headers on-chain. The callback is fired when new headers
   // are found, the call itself returns a promise with a subscription that can be
-  // used to unsubscribe from the newHead subscription
+  // used to unsubscribe from the newHead subscription.
+  // Use the RPC Node Interface
   const subscriptionId = await api.rpc.chain.newHead((header) => {
     console.log(`best #${header.blockNumber}`);
   });
 
   // Id for the subscription, we can cleanup and unsubscribe via
   // `api.chain.newHead.unsubscribe(subscriptionId)`
-  console.log(`subsciptionId: ${subscriptionId}`);
+  console.log(`subscriptionId: ${subscriptionId}`);
+
+  setTimeout(() => {
```

Comment:
Ok, so we will never see more than 1 or 2 blocks? Not quite the intent of the example at all :(

Reply:
I believe we just keep it running until the user exits. No need to add delays, we are trying to show how things work and allow people to experiment. Plenty of other "unsub" examples.